Fix/exception handler

MEHRSHAD-MIRSHEKARY opened the pull request.

MEHRSHAD-MIRSHEKARY committed:
fix(exceptions): `extract_first_error` returns string for dict-type errors

Fixed an issue in extract_first_error where dictionary-type errors were incorrectly returned as a dictionary instead of a string. Now, when encountering a d
ictionary, the method properly extracts and returns the first string error message.

**Changes**:

 - Updated dictionary handling logic to recursively extract and return a string instead of a {key: value} structure.
 - Ensured consistency in error extraction for better compatibility with validation error responses.

Closes #34

ARYAN-NIKNEZHAD commented on response_shaper/exceptions.py:
```diff
@@ -161,32 +161,30 @@ def db_error(exc, status=500):
         if isinstance(error_data, dict):
-            for key, value in error_data.items():
-                return {key: ExceptionHandler.extract_first_error(value)}
+            for value in error_data.values():
```
Keep this dynamic in the settings, allowing users to choose between options (either as a string or a JSON dictionary).

MEHRSHAD-MIRSHEKARY replied:
I have modified the logic and applied the requested changes

MEHRSHAD-MIRSHEKARY committed:
refactor(exceptions): Enhance `extract_first_error` for nested dicts to return one key-value pair

Add new config named `RETURN_ERROR_AS_DICT` for controling the error response, making it more intuitive that it toggles between dict and string output.
 Modify the ExceptionHandler.extract_first_error method to better handle
three or more nested dictionaries. `When RETURN_ERROR_AS_DICT` is True,
it now returns the innermost key-value pair as a single-level dict
(e.g., {"c": "error message"} from {"a": {"b": {"c": "error message"}}})
instead of the full nested structure. When False, it continues to return
the innermost string value.

Update includes:
- Recursive logic adjustment to target innermost dict with string value
- Support for arbitrary nesting levels

MEHRSHAD-MIRSHEKARY committed:
:books: docs(README): Update readme documentation to add new config
